src/NServiceBus.RavenDB/Timeouts/TimeoutPersister.cs

```diff
@@ -16,55 +16,54 @@ public TimeoutPersister(IDocumentStore store)
             documentStore = store;
         }
 
-        public Task Add(CoreTimeoutData timeout, ContextBag context)
+        public async Task Add(CoreTimeoutData timeout, ContextBag context)
         {
-            using (var session = documentStore.OpenSession())
+            using (var session = documentStore.OpenAsyncSession())
             {
-                session.Store(new Timeout(timeout));
-                session.SaveChanges();
+                await session.StoreAsync(new Timeout(timeout)).ConfigureAwait(false);
+                await session.SaveChangesAsync().ConfigureAwait(false);
             }
-            return Task.FromResult(0);
         }
 
-        public Task<bool> TryRemove(string timeoutId, ContextBag context)
+        public async Task<bool> TryRemove(string timeoutId, ContextBag context)
         {
-            using (var session = documentStore.OpenSession())
+            using (var session = documentStore.OpenAsyncSession())
             {
                 session.Advanced.UseOptimisticConcurrency = true;
 
-                var timeout = session.Load<Timeout>(timeoutId);
+                var timeout = await session.LoadAsync<Timeout>(timeoutId).ConfigureAwait(false);
```

Comment:
there is no direct delete operation on raven which also returns true/false, right?